Add procedure for Installable Applicable Errata (#2295)
theforeman/foreman-documentation: guides/common/modules/proc_generating-a-report-for-installable-and-applicable-errata.adoc

[id="Generating_a_Report_for_Installable_and_Applicable_Errata_{context}"]
= Generating a Report for Installable and Applicable Errata

Use the following procedure to view a list of installable and applicable errata to install.

.Procedure
. In the {ProjectWebUI}, navigate to *Monitor* > *Report Templates*.

> . To the right of the report template that you want to use, click *Generate*.

Reviewer: ```suggestion
. For the *Host {endash} Applicable Errata* template, click *Generate*.
```

Still not happy with "For the ..."; maybe you can find a better wording here.

Author: Reworded it to `. Click *Generate* for the *Host {endash} Applicable Errata* template.`

. Select *Applicable* to show all applicable errata.
. Select *Installable*  to limit the report exclusively to errata that are accessible in the host's content view environment(s) that can be installed.
. Optional: To schedule a report, click the icon to the right of the *Generate at* field and choose the date and time you want for the generated report.
. Optional: To send a report to an e-mail address, select the *Send report via e-mail* checkbox, and in the *Deliver to e-mail addresses* field, enter the required e-mail address.

> A CSV file that contains the report is downloaded.

Reviewer: ```suggestion
. Click *Generate* to generate a CSV file that contains the report.
After {Project} has created the report, your browser automatically downloads it.
```

> If you have selected the *Send report via e-mail* checkbox, the host monitoring report is sent to your e-mail address.

Reviewer: ```suggestion
+
If you have selected the *Send report via e-mail* checkbox, the report is sent to your e-mail address.
```

[id="Generating_a_Report_for_Installable_and_Applicable_Errata_{context}"]
= Generating a Report for Installable and Applicable Errata

Use the following procedure to view installable and applicable errata on managed hosts.

.Procedure
. In the {ProjectWebUI}, navigate to *Monitor* > *Report Templates*.
. Click *Generate* for the *Host {endash} Applicable Errata* template.
. In the *Installability* drop down menu, select *Applicable* to show all applicable errata or *Installable* to limit the report exclusively to errata that are accessible in the Content View environments of your host that can be installed.
. Optional: To schedule a report, click the icon to the right of the *Generate at* field and choose the date and time you want for the generated report.

> . Optional: To send a report to an e-mail address, select the *Send report via e-mail* checkbox, and in the *Deliver to e-mail addresses* field, enter the required e-mail address.

Reviewer 2: For these two Optional steps, I would suggest not to describe the UI in so much detail.

. Optional: To schedule a report, use the *Generate at* field.
. Optional: To send a report to an e-mail address, select the *Send report via e-mail* checkbox and then enter the required e-mail address.

This is because UIs can change a lot, and the more details we describe, the more updating there is later. In general, I find this particular web UI very well designed and easily understandable even without detailed docs.

But this is just a suggestion. If you experience says otherwise, feel free to ignore it :)

Author: I actually agree with you. I did, however, copy this from **Generating Host Monitoring Reports** in the _Managing Hosts_ guide. :-)

I'll keep it as is for now and then after this is merged, maybe we change it here and in the _Managing Hosts_ guide in a separate PR.

. Click *Generate* to generate a CSV file that contains the report.
After {Project} has created the report, your browser automatically downloads it.
+

> If you have selected the *Send report via e-mail* checkbox, the report is sent to your e-mail address.

Reviewer 2: I would avoid using present perfect tense here and on line 13 if it makes sense to you.

[id="Generating_a_Report_for_Installable_and_Applicable_Errata_{context}"]
= Generating a Report for Installable and Applicable Errata

> Use the following procedure to view a list of installable and applicable errata to install.

Reviewer: ```suggestion
Use the following procedure to view installable and applicable errata on managed hosts.
```

.Procedure
. In the {ProjectWebUI}, navigate to *Monitor* > *Report Templates*.
. Click *Generate* for the *Host {endash} Applicable Errata* template.

> . Select *Applicable* to show all applicable errata.
> . Select *Installable* to limit the report exclusively to errata that are accessible in the content view environments of your host that can be installed.

Reviewer: ```suggestion
. In the *Instalability* drop down menu, select *Applicable* to show all applicable errata or *Installable* to limit the report exclusively to errata that are accessible in the content view environments of your host that can be installed.
```

[id="Generating_a_Report_for_Installable_and_Applicable_Errata_{context}"]
= Generating a Report for Installable and Applicable Errata

Use the following procedure to generate a report of installable or applicable errata on managed hosts.

.Procedure
. In the {ProjectWebUI}, navigate to *Monitor* > *Report Templates*.
. Click *Generate* for the *Host {endash} Applicable Errata* template.

> . In the *Installability* drop down menu, select one of these options:

Reviewer 3: ```suggestion
. In the *Installability* drop-down menu, select one of these options:
```

Reviewer 3: Just a minor.

* *Applicable* to show all applicable errata.
* *Installable* to limit the report exclusively to errata that are accessible in the Content View environments of your host that may be installed.

> . Optional: To schedule a report, click the icon to the right of the *Generate at* field and choose the date and time you want for the generated report.

Reviewer: ```suggestion
. Optional: To schedule a report, click the *calendar* icon to the right of the *Generate at* field and choose the date and time you want for the generated report.
```

[id="Generating_a_Report_for_Installable_and_Applicable_Errata_{context}"]
= Generating a Report for Installable and Applicable Errata

Use the following procedure to generate a report of installable or applicable errata on managed hosts.

.Procedure
. In the {ProjectWebUI}, navigate to *Monitor* > *Report Templates*.
. Click *Generate* for the *Host {endash} Applicable Errata* template.
. Optional: To schedule a report, click the *calendar* icon to the right of the *Generate at* field and choose the date and time you want for the generated report.
. Optional: To send a report to an e-mail address, select the *Send report via e-mail* checkbox, and in the *Deliver to e-mail addresses* field, enter the required e-mail address.
. Optional: To limit the report only to hosts found by the search query, click on *Hosts filter* and search from the available list of hosts.
For a report on all available hosts, leave *Hosts filter* empty.
. Optional: To limit the report only to errata found by the search query, click on *Errata filter* and search from the available list of errata.
For a report on all available errata, leave *Errata filter* empty.

> . In the *Installability* drop-down menu, select one of these options:

Reviewer 4: ```suggestion
. From the *Installability* list, select one of these options:
```
According to the IBM style guide, we shouldn't use "drop down".

* *Applicable* to show all applicable errata.
* *Installable* to limit the report exclusively to errata that are accessible in the Content View environments of your host that may be installed.

> . Click *Generate* to generate a CSV file that contains the report.

Reviewer 4: What about the option to select the output format of the report?

[id="Generating_a_Report_for_Installable_and_Applicable_Errata_{context}"]
= Generating a Report for Installable and Applicable Errata

Use the following procedure to generate a report of installable or applicable errata on managed hosts.

.Procedure
. In the {ProjectWebUI}, navigate to *Monitor* > *Report Templates*.
. Click *Generate* for the *Host {endash} Applicable Errata* template.
. Optional: To schedule a report, click the *calendar* icon to the right of the *Generate at* field and choose the date and time you want for the generated report.

> . Optional: To send a report to an e-mail address, select the *Send report via e-mail* checkbox, and in the *Deliver to e-mail addresses* field, enter the required e-mail address.

Reviewer 4: ```suggestion
. Optional: To send a report to an e-mail address, select the *Send report via e-mail* checkbox, and in the *Deliver to e-mail addresses* field, enter the required e-mail address.
. Optional: Select another *Output format* for the report file.
The default is CSV.
```

. Optional: To limit the report only to hosts found by the search query, click on *Hosts filter* and search from the available list of hosts.
For a report on all available hosts, leave *Hosts filter* empty.
. Optional: To limit the report only to errata found by the search query, click on *Errata filter* and search from the available list of errata.
For a report on all available errata, leave *Errata filter* empty.
. From the *Installability* list, select one of these options:
* *Applicable* to show all applicable errata.
* *Installable* to limit the report exclusively to errata that are accessible in the Content View environments of your host that may be installed.

> . Click *Generate* to generate a file that contains the report.
> The default file is CSV and the other options to choose from are JSON, YAML, and HTML.
> Your browser automatically downloads the report after {Project} creates it.
> +
> If you have selected the *Send report via e-mail* checkbox, the report is sent to your e-mail address.

Reviewer 4: ```suggestion
. Click *Generate*.
Your browser automatically downloads the report file after {Project} creates it.
If you have selected the *Send report via e-mail* option, the report is sent to your e-mail address.
```
